src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -391,6 +407,23 @@ private static string CombineNoChecks(ReadOnlySpan<char> first, ReadOnlySpan<cha
             return CombineNoChecksInternal(first, second, third);            
         }
 
+        private static string CombineNoChecks(string first, string second, string third)
+        {
+            if (first.Length == 0)
+                return CombineNoChecks(second, third);
+            if (second.Length == 0)
+                return CombineNoChecks(first, third);
+            if (third.Length == 0)
+                return CombineNoChecks(first, second);
+
+            if (IsPathRooted(third.AsReadOnlySpan()))
```

Comment:
why explicitly convert to ReadOnlySpan?

Reply:
I did this because ispathrooted(string) will be called which will call ispathrooted(span). As we have already done the null check, explicit conversion will avoid a function call.

Reply:
should I keep it or just call the string version ?

Reply:
No, it is fine to leave as is.